### ISSUE TITLE
AWS SES Node: Malformed Query String

### ISSUE THREAD

**Issue by coltoneakins** [n8n-io/n8n#1442]
## **Describe the bug**

When using the AWS SES node (to send emails), data sent to AWS is not escaped.

I am trying to send emails that contain HTML (as part of the data) to users. So, the email body contains commonly used characters in HTML such as "=" (like for example, sending a link in the body of the email).

I do have the 'Is Body HTML?' option checked ON for this node. I would expect this to properly escape HTML email bodies--but it is not.

Here is the body of the message I am sending:

```
<p>This is a test message body with some html: <a href="https://example.com">link</a>.</p>
```

Exact error: (Note: personal information was replaced with 'mywebsite.com' and 'example.com')

```
ERROR: AWS error response [400]: 400 - "<ErrorResponse xmlns=\"http://webservices.amazon.com/AWSFault/2005-15-09\">\n <Error>\n <Type>Sender</Type>\n <Code>MalformedQueryString</Code>\n <Message>Action=SendEmail&amp;Message.Subject.Data=Test&amp;Source=notifications@mywebsite.com&amp;Message.Body.Html.Data=%3Cp%3EThis%20is%20a%20test%20message%20body%20with%20some%20html:%20%3Ca%20href=%22https://example.com%22%3Elink%3C/a%3E.%3C/p%3E&amp;Destination.ToAddresses.member.1=email@example.com is not valid; the value of a query string parameter may not contain a '=' delimiter</Message>\n </Error>\n <RequestId>c279f093-f58a-4f2a-a232-fae19985b892</RequestId>\n</ErrorResponse>\n"
Error: AWS error response [400]: 400 - "<ErrorResponse xmlns=\"http://webservices.amazon.com/AWSFault/2005-15-09\">\n  <Error>\n    <Type>Sender</Type>\n    <Code>MalformedQueryString</Code>\n    <Message>Action=SendEmail&amp;Message.Subject.Data=Test&amp;Source=notifications@mywebsite.com&amp;Message.Body.Html.Data=%3Cp%3EThis%20is%20a%20test%20message%20body%20with%20some%20html:%20%3Ca%20href=%22https://example.com%22%3Elink%3C/a%3E.%3C/p%3E&amp;Destination.ToAddresses.member.1=email@example.com is not valid; the value of a query string parameter may not contain a '=' delimiter</Message>\n  </Error>\n  <RequestId>c279f093-f58a-4f2a-a232-fae19985b892</RequestId>\n</ErrorResponse>\n"
    at Object.awsApiRequest (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Aws/SES/GenericFunctions.js:35:15)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Object.awsApiRequestSOAP (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Aws/SES/GenericFunctions.js:50:22)
    at async Object.execute (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Aws/SES/AwsSes.node.js:1049:36)
    at async /usr/local/lib/node_modules/n8n/node_modules/n8n-core/dist/src/WorkflowExecute.js:395:47

```

**When the email is sent to AWS, AWS says the query string is malformed. This is because the '=' equals sign in my email body is not being escaped (or URL encoded as %3D).**

##  **To Reproduce**
1. Create an AWS SES node in N8N. You will have to set up credentials via the AWS console.
2. Go to the setting for the AWS SES node. Set the node to the following:
  - Resource: Email
  - Operation: Send
  - Is Body HTML: Checked ON
  - Subject: Test
  - Body: `<p>This is a test message body with some html: <a href="https://example.com">link</a>.</p>`
  - From Email: {{YOUR-AWS-VERIFIED-MAIL-FROM-EMAIL}}
  - To addresses: {{YOUR-AWS-VERIFIED-RECIPIENT-EMAIL}}
3. Once the node is properly set up w/ AWS SES, use the 'Execute Node' feature to attempt to send an email. AWS SES will spit back an error back: MalformedQueryString. (Because the email body contains a '=' equals sign--which is not allowed in query strings.) 

## **Expected behavior**
The email is sent with HTML properly escaped in the email body. AWS will reject messages containing a bad query string.

## **Environment (please complete the following information):**

I am using the exact Docker setup described here: https://docs.n8n.io/reference/server-setup.html

I am using the latest docker image. All of the information below is taken from within the Docker container (so that it is accurate).

 - OS: Amazon Linux 2 w/ Docker
 - n8n Version: 0.107.0
 - Node.js Version: v14.15.4

## **Additional context**
No other context. I described the issue thoroughly above.

**Comment by coltoneakins** [comment]
For those looking for an alternative (until this bug is fixed), you can use the core 'Email' node with SES. AWS SES allows you to set up SMTP credentials that can be used with the 'Email' node.

**Comment by krynble** [comment]
Hi @coltoneakins 

Sorry about the delay on this. I just opened a pull request that solves the problem. Should be released on the next version.

**Comment by spessex** [comment]
hi @krynble.

That's uncanny, as I've been experiencing the same issue for the past couple of days and have had to resort to the "Send Email' node for now (which works but I'm having a few issues on formatting differences in Desktop and Mobile - where in Desktop all LInks/URL are blue/clickable, yet mobile is just plain text). Regardless, great to hear of the fix, but because I'm new to n8n I just wanted to know when this might be released? You mention on the next version, but how are versions usually scheduled?

Thanks again

Stephen

**Comment by janober** [comment]
@spessex Glad to hear that this also fixed your problem. We release a new version every week. The next one will probably be released next Sunday.

**Comment by spessex** [comment]
Fabulous. Great news. Thank you.

**Comment by janober** [comment]
Got released with n8n@0.134.0

**Comment by bamboowonder** [comment]
This seems to be broken again.   https://community.n8n.io/t/amazon-ses-error-on-valid-body-subject/11116/5
